gcloud-java-dns/src/test/java/com/google/cloud/dns/it/ITDnsTest.java

```diff
@@ -967,25 +969,920 @@ public void testListDnsRecords() {
+          Dns.ChangeRequestListOption.pageSize(0));
+      DnsBatchResult<Page<ChangeRequest>> errorPageNegative = batch.listChangeRequests(ZONE1.name(),
+          Dns.ChangeRequestListOption.pageSize(-1));
+      DnsBatchResult<Page<ChangeRequest>> resultAscending = batch.listChangeRequests(ZONE1.name(),
+          Dns.ChangeRequestListOption.sortOrder(Dns.SortingOrder.ASCENDING));
+      DnsBatchResult<Page<ChangeRequest>> resultDescending = batch.listChangeRequests(ZONE1.name(),
+          Dns.ChangeRequestListOption.sortOrder(Dns.SortingOrder.DESCENDING));
+      DnsBatchResult<Page<ChangeRequest>> resultAdditions = batch.listChangeRequests(ZONE1.name(),
+          Dns.ChangeRequestListOption.sortOrder(Dns.SortingOrder.ASCENDING),
+          Dns.ChangeRequestListOption.fields(ChangeRequestField.ADDITIONS));
+      DnsBatchResult<Page<ChangeRequest>> resultDeletions = batch.listChangeRequests(ZONE1.name(),
+          Dns.ChangeRequestListOption.sortOrder(Dns.SortingOrder.ASCENDING),
+          Dns.ChangeRequestListOption.fields(ChangeRequestField.DELETIONS));
+      DnsBatchResult<Page<ChangeRequest>> resultId = batch.listChangeRequests(ZONE1.name(),
+          Dns.ChangeRequestListOption.sortOrder(Dns.SortingOrder.ASCENDING),
+          Dns.ChangeRequestListOption.fields(ChangeRequestField.ID));
+      DnsBatchResult<Page<ChangeRequest>> resultTime = batch.listChangeRequests(ZONE1.name(),
+          Dns.ChangeRequestListOption.sortOrder(Dns.SortingOrder.ASCENDING),
+          Dns.ChangeRequestListOption.fields(ChangeRequestField.START_TIME));
+      DnsBatchResult<Page<ChangeRequest>> resultStatus = batch.listChangeRequests(ZONE1.name(),
+          Dns.ChangeRequestListOption.sortOrder(Dns.SortingOrder.ASCENDING),
+          Dns.ChangeRequestListOption.fields(ChangeRequestField.STATUS));
+      batch.submit();
+      changes = ImmutableList.copyOf(result.get().iterateAll());
+      assertEquals(3, changes.size());
+      // error in options
+      try {
+        errorPageSize.get();
+        fail();
+      } catch (DnsException ex) {
+        // expected
+        assertEquals(400, ex.code());
+        assertFalse(ex.retryable());
+      }
+      try {
+        errorPageNegative.get();
+        fail();
+      } catch (DnsException ex) {
+        // expected
+        assertEquals(400, ex.code());
+        assertFalse(ex.retryable());
+      }
+      // sorting order
+      ImmutableList<ChangeRequest> ascending =
+          ImmutableList.copyOf(resultAscending.get().iterateAll());
+      ImmutableList<ChangeRequest> descending =
+          ImmutableList.copyOf(resultDescending.get().iterateAll());
+      int size = 3;
+      assertEquals(size, descending.size());
+      assertEquals(size, ascending.size());
+      for (int i = 0; i < size; i++) {
+        assertEquals(descending.get(i), ascending.get(size - i - 1));
+      }
+      // field options
+      changes = ImmutableList.copyOf(resultAdditions.get().iterateAll());
```

Comment:
I think that in `testListZonesBatch` you use the iterator to go through the results. I slightly prefer that to avoid copies. (also applies to the following)